File: Python/libraries/recognizers-sequence/recognizers_sequence/sequence/config/url_configuration.py
from recognizers_sequence.sequence.extractors import *
from recognizers_sequence.sequence.config.sequence_configuration import SequenceConfiguration


class URLConfiguration(BaseSequenceExtractorConfiguration):

    def __init__(self, ip_url_regex, url_regex, options: SequenceConfiguration):
        self.__ip_url_regex = ip_url_regex
        self.__url_regex = url_regex
        self.options = options

    @property
    def ip_url_regex(self) -> str:
        return self.__ip_url_regex

    @ip_url_regex.setter
    def word_boundaries_regex(self, ip_url_regex):
        self.__ip_url_regex = ip_url_regex

    @property
    def url_regex(self) -> str:
        return self.__url_regex

    @url_regex.setter
    def url_regex(self, url_regex):
        self.__url_regex = url_regex

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/config/URLConfiguration.java
package com.microsoft.recognizers.text.sequence.config;

import com.microsoft.recognizers.text.sequence.SequenceOptions;

import java.util.regex.Pattern;

public class URLConfiguration implements ISequenceConfiguration {
    private SequenceOptions options;
    private Pattern ipUrlRegex;
    private Pattern urlRegex;

    public URLConfiguration(SequenceOptions options) {
        this.options = options != null ? options : SequenceOptions.None;
    }

    @Override
    public SequenceOptions getOptions() {
        return this.options;
    }

    public Pattern getIpUrlRegex() {
        return ipUrlRegex;
    }

    public void setIpUrlRegex(Pattern ipUrlRegex) {
        this.ipUrlRegex = ipUrlRegex;
    }

    public Pattern getUrlRegex() {
        return urlRegex;
    }

    public void setUrlRegex(Pattern urlRegex) {
        this.urlRegex = urlRegex;
    }
}
